_posts/2019-09-05-cover-song-similarity.md

```diff
@@ -0,0 +1,263 @@
+
+```python
+csm_oti = estd.ChromaCrossSimilarity(frameStackSize=9, 
+                                    frameStackStride=1, 
+                                    binarizePercentile=0.095,
+                                    oti=True,
+                                    otiBinary=True)
+
+oti_csm = csm_oti(query_hpcp, false_cover_hpcp)
+```
+
+
+#### Step 3
+
+The last step is to compute an cover song similarity distance between both `query-true_cover` and `query-false_cover` pairs using local sub-sequence alignment algorithm designed for cover song identification task. [`CoverSongSimilarity`](https://essentia.upf.edu/reference/std_CoverSongSimilarity.html) algorithm in essentia provides two alignment constraints of smith-waterman algorithm [6] based on [2] and [3]. We can switch between these methods using the `alignmentType` parameter. Please refer the documentation for more details.
+
+
+Let's compute the cover song similarity distance between true cover song pairs.
+
+```python
+score_matrix, distance = estd.CoverSongSimilarity(disOnset=0.5, 
+                                            disExtension=0.5, 
+                                            alignmentType='serra09',
+                                            distanceType='asymmetric')
+                                            (true_pair_crp)
+
+# plot
+plt.title('Cover song similarity distance: %s' % distance)
+plt.xlabel('Yesterday accapella cover')
+plt.ylabel('Yesterday - The Beatles')
+plt.imshow(score_matrix, origin='lower')
+```
+![png]({{ site.baseurl }}/assets/cover-song-similarity/true-cover-pair-qmax.png)
+
+
+
+Now, let's compute the cover song similarity distance between false cover song pairs.
+
+```python
+score_matrix, distance = estd.CoverSongSimilarity(disOnset=0.5, 
+                                            disExtension=0.5, 
+                                            alignmentType='serra09',
+                                            distanceType='asymmetric')
+                                            (false_pair_crp)
+
+# plot
+plt.title('Cover song similarity distance: %s' % distance)
+plt.ylabel('Yesterday accapella cover')
+plt.xlabel('Come together cover - Aerosmith')
+plt.imshow(score_matrix, origin='lower')
+```
+
+![png]({{ site.baseurl }}/assets/cover-song-similarity/false-cover-pair-qmax.png)
+
+
+Voila! We can see that the cover similarity distance is quite low for the actual cover song pairs as expected.
+
+
+## Streaming mode computation
+
+We can also compute the aforementioned similarity measures in essentia streaming mode which suits its real-time application use cases. The following code block shows an simple example of the workflow where we stream the query song HPCP feature to compute the real-time cover similarity distance between a pre-selected reference song HPCP feature.
```

Comment:
"We can also compute the same similarity measures using the streaming mode, which suits real-time applications."